### PR TITLE
Move the release process to GitHub Actions

### PR DESCRIPTION
**Pull request by SajidAlamQB**
## Description

Context: #2969 

An example working flow can be seen on my personal repository: https://github.com/SajidAlamQB/kedro1
Test PyPi: https://test.pypi.org/project/kedro-test-pypi/0.19.1/

Similar to kedro plugins we need to add repository secrets for github: `GH_TAGGING_TOKEN` and pypi: `KEDRO_PYPI_TOKEN`

## Development notes

Took most of the workflow example from https://github.com/kedro-org/kedro-plugins/blob/main/.github/workflows/check-release.yml and adapted it for kedro.

Removed all CircleCI related release scripts.

Included the automatic GitHub tag and release creation with release notes.

## Developer Certificate of Origin
We need all contributions to comply with the [Developer Certificate of Origin (DCO)](https://developercertificate.org/). All commits must be signed off by including a `Signed-off-by` line in the commit message. [See our wiki for guidance](https://github.com/kedro-org/kedro/wiki/Guidelines-for-contributing-developers/).

If your PR is blocked due to unsigned commits, then you must follow the instructions under "Rebase the branch" on the GitHub Checks page for your PR. This will retroactively add the sign-off to all unsigned commits and allow the DCO check to pass.

## Checklist

- [x] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [x] Signed off each commit with a [Developer Certificate of Origin (DCO)](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-the-commit-signoff-policy-for-your-repository)
- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes
- [ ] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

File: .github/workflows/check-release.yml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+      - name: Extract release notes from RELEASE.md
+        id: extract
+        run: |
+          python tools/github_actions/extract_release_notes.py \
+          "RELEASE.md" \
+          "Release ${{needs.check-version.outputs.kedro_version}}"
+      - name: Create GitHub Release
+        uses: softprops/action-gh-release@v1
+        with:
+          tag_name: kedro-${{needs.check-version.outputs.kedro_version}}
```
Comment (ankatiyar) [inline .github/workflows/check-release.yml:66]: I think the tags we use on Kedro is simply the version number and not `kedro-<version>`

File: .github/workflows/check-release.yml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+          name: kedro-${{needs.check-version.outputs.kedro_version}}
```
Comment (ankatiyar) [inline .github/workflows/check-release.yml:67]: The name should also just be the version number according to the previous releases - https://github.com/kedro-org/kedro/releases

**Review by ankatiyar — APPROVED** [review]
Thanks @SajidAlamQB ⭐ ! I left some comments about the Github release step but overall looks good to me!

**Comment by merelcht** [comment]
`GH_TAGGING_TOKEN` is my personal token, so that should just work fine without changes. I've created `KEDRO_PYPI_TOKEN`.

**Review by merelcht — APPROVED** [review]
LGTM! ⭐ Great to see so many `cci` files go already. Not long until we can remove everything!

**Pull request merged**